### PR TITLE
Reduce column mapping mode usage in CDF tests

### PR DESCRIPTION
**Pull request by ebyhr**
## Description

The remaining test cases cover column mapping mode with UPDATE, DELETE, MERGE, non-lowercase columns and partitioned table.

Relates to #18805

## Release notes

(x) This is not user-visible or is docs only, and no release notes are required.

### CODE REVIEW

**Review by findinpath — APPROVED** [review]

File: testing/trino-product-tests/src/main/java/io/trino/tests/product/deltalake/TestDeltaLakeChangeDataFeedCompatibility.java [hunk 1]
```diff
@@ -452,17 +451,17 @@ public void testMergeDeleteIntoTableWithCdfEnabled(String columnMappingMode)
         }
     }
 
-    @Test(groups = {DELTA_LAKE_DATABRICKS, DELTA_LAKE_OSS, DELTA_LAKE_EXCLUDE_91, PROFILE_SPECIFIC_TESTS}, dataProvider = "columnMappingModeDataProvider")
+    @Test(groups = {DELTA_LAKE_DATABRICKS, DELTA_LAKE_OSS, DELTA_LAKE_EXCLUDE_91, PROFILE_SPECIFIC_TESTS})
     @Flaky(issue = DATABRICKS_COMMUNICATION_FAILURE_ISSUE, match = DATABRICKS_COMMUNICATION_FAILURE_MATCH)
-    public void testMergeMixedDeleteAndUpdateIntoTableWithCdfEnabled(String columnMappingMode)
+    public void testMergeMixedDeleteAndUpdateIntoTableWithCdfEnabled()
```
Comment (homar) [inline testing/trino-product-tests/src/main/java/io/trino/tests/product/deltalake/TestDeltaLakeChangeDataFeedCompatibility.java:456]: I'd leave this test as it is,I dion't see this case being covered anywhere else

**Review by findepi — APPROVED** [review]

**Comment by findepi** [comment]
please mind @homar 's concerns

File: testing/trino-product-tests/src/main/java/io/trino/tests/product/deltalake/TestDeltaLakeChangeDataFeedCompatibility.java (hunk 1 above)
Comment (ebyhr) [reply testing/trino-product-tests/src/main/java/io/trino/tests/product/deltalake/TestDeltaLakeChangeDataFeedCompatibility.java:456]: Do you think `TestDeltaLakeConnectorTest#testReadMergeChanges` is insufficient?

Comment (homar) [reply testing/trino-product-tests/src/main/java/io/trino/tests/product/deltalake/TestDeltaLakeChangeDataFeedCompatibility.java:456]: oh i forgot about that one, sure it is fine

**Review by homar — APPROVED** [review]

**Pull request merged**